File: backend/app/schemas/medical_history.py
"""
Medical and Family History Schemas

Defines request and response shapes for medical and family history API data.
"""

from datetime import date

from pydantic import BaseModel


class MedicalHistoryBase(BaseModel):
    condition: str
    diagnosis_date: date | None = None
    notes: str | None = None


class MedicalHistoryCreate(MedicalHistoryBase):
    pass


class MedicalHistory(MedicalHistoryBase):
    medical_history_id: int
    patient_id: int

    class Config:
        from_attributes = True


class FamilyHistoryBase(BaseModel):
    relationship: str
    condition: str
    notes: str | None = None


class FamilyHistoryCreate(FamilyHistoryBase):
    pass


class FamilyHistory(FamilyHistoryBase):
    family_history_id: int
    patient_id: int

    class Config:
        from_attributes = True

File: frontend/src/api/medicalHistory.ts
import type { FamilyHistory, MedicalHistory } from "../types";
import { apiFetch } from "./client";

export interface CreateMedicalHistoryInput {
  condition: string;
  diagnosis_date: string | null;
  notes: string | null;
}

export interface CreateFamilyHistoryInput {
  relationship: string;
  condition: string;
  notes: string | null;
}

export const medicalHistoryApi = {
  listMedical: (patientId: number) =>
    apiFetch<MedicalHistory[]>(`/patients/${patientId}/medical-history`),

  createMedical: (patientId: number, input: CreateMedicalHistoryInput) =>
    apiFetch<MedicalHistory>(`/patients/${patientId}/medical-history`, {
      method: "POST",
      body: JSON.stringify(input),
    }),

  removeMedical: (patientId: number, medicalHistoryId: number) =>
    apiFetch<void>(
      `/patients/${patientId}/medical-history/${medicalHistoryId}`,
      { method: "DELETE" }
    ),

  listFamily: (patientId: number) =>
    apiFetch<FamilyHistory[]>(`/patients/${patientId}/family-history`),

  createFamily: (patientId: number, input: CreateFamilyHistoryInput) =>
    apiFetch<FamilyHistory>(`/patients/${patientId}/family-history`, {
      method: "POST",
      body: JSON.stringify(input),
    }),

  removeFamily: (patientId: number, familyHistoryId: number) =>
    apiFetch<void>(
      `/patients/${patientId}/family-history/${familyHistoryId}`,
      { method: "DELETE" }
    ),
};